test/unit/math/prim/fun/fft_test.cpp

```diff
@@ -0,0 +1,208 @@
+#include <stan/math/prim.hpp>
+#include <test/unit/util.hpp>
+#include <gtest/gtest.h>
+
+TEST(primFun, fft) {
+  typedef std::complex<double> c_t;
+  typedef Eigen::Matrix<std::complex<double>, -1, 1> cv_t;
+  using stan::math::fft;
+
+  // reference answers calculated using Scipy.fft with double precision
+
+  cv_t x0(0);
+  cv_t y0 = fft(x0);
+  EXPECT_EQ(0, y0.size());
+
+  cv_t x1(1);
+  x1 << c_t(-3.247, 1.98555);
+  cv_t y1 = fft(x1);
+  EXPECT_EQ(1, y1.size());
+  EXPECT_EQ(real(y1[0]), -3.247);
+  EXPECT_EQ(imag(y1[0]), 1.98555);
+
+  cv_t y1b = fft(x1 + x1);
+  EXPECT_NEAR(real(y1b[0]), -3.247 * 2, 1e-6);
+  EXPECT_NEAR(imag(y1b[0]), 1.98555 * 2, 1e-6);
+
+  cv_t x(3);
+  x << c_t(1, -2), c_t(-3, 5), c_t(-7, 11);
+  cv_t y = fft(x);
+  EXPECT_EQ(3, y.size());
+  EXPECT_NEAR(real(y[0]), -9, 1e-6);
+  EXPECT_NEAR(imag(y[0]), 14, 1e-6);
+  EXPECT_NEAR(real(y[1]), 0.80384758, 1e-6);
+  EXPECT_NEAR(imag(y[1]), -13.46410162, 1e-6);
+  EXPECT_NEAR(real(y[2]), 11.19615242, 1e-6);
+  EXPECT_NEAR(imag(y[2]), -6.53589838, 1e-6);
```

Comment:
Is `EXPECT_NEAR` needed because of differences in the python and eigen versions of fft? This is essentially just testing Eigen's impl of fft right? imo might be better to just use values from Eigen's for the test then this test scheme just becomes a canary for when we update eigen versions to see if any changes happened.

Reply:
They need floating-point comparison.  I replaced explicit `EXPECT_NEAR` with implicit `EXPECT_FLOAT_EQ`, and it all still works, so I'll leave that.